pkg/policy/repository.go

```diff
@@ -777,11 +788,23 @@ func (p *Repository) computePolicyEnforcementAndRules(securityIdentity *identity
 	}
 
 	matchingRules = []*rule{}
-	for _, r := range p.rules {
+	// Match cluster-wide rules
+	for rKey := range p.rulesByNamespace[""] {
+		r := p.rules[rKey]
 		if r.matchesSubject(securityIdentity) {
 			matchingRules = append(matchingRules, r)
 		}
 	}
+	// Match namespace-wide rules
```

Comment:
This would read better as "Match namespace-specific rules"

Reply:
done and done.